lib/srv/authhandlers.go

```diff
@@ -159,6 +159,14 @@ func (h *AuthHandlers) CheckAgentForward(ctx *ServerContext) error {
 	return nil
 }
 
+// CheckX11Forward checks if X11 forwarding is permitted for the user's RoleSet.
+func (h *AuthHandlers) CheckX11Forward(ctx *ServerContext) error {
+	if !ctx.Identity.RoleSet.PermitX11Forwarding() {
+		return trace.Wrap(trace.AccessDenied("x11 forwarding not permitted"))
```

Comment:
1. The other checkers return `trace.AccessDenied` directly, rather than wrapping it. Should you do the same?
2. Should you emit an audit event when access to X11 forwarding is denied?